sql/core/src/main/scala/org/apache/spark/sql/execution/Generate.scala

```diff
@@ -75,9 +78,15 @@ case class Generate(
             outputRows.map(or => joinProjection(joinedRow(row, or)))
           }
         }
+      }.mapPartitions { iter =>
+        val nullOriginalRow = Row(Seq.fill(generator.output.size)(Literal(null)): _*)
```

Comment:
Yes, you're right, we should use the `child.output.size`, not the `generator.output.size`.